ci-operator/step-registry/gcp/provision/cco-manual-users/static/gcp-provision-cco-manual-users-static-commands.sh

```diff
@@ -0,0 +1,233 @@
+# - storage.objects.list
+##
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+MPREFIX="${SHARED_DIR}/manifest"
+creds_requests_dir="$(mktemp -d)"
+creds_keys_dir="$(mktemp -d)"
+creds_manifests_dir="$(mktemp -d)"
+
+echo "$(date -u --rfc-3339=seconds) - Extracting GCP credentials requests from the release image..."
+ADDITIONAL_OC_EXTRACT_ARGS=""
+if [[ "${EXTRACT_MANIFEST_INCLUDED}" == "true" ]]; then
+  ADDITIONAL_OC_EXTRACT_ARGS="${ADDITIONAL_OC_EXTRACT_ARGS} --included --install-config=${SHARED_DIR}/install-config.yaml"
+fi
+echo "OC Version:"
+which oc
+oc version --client
+oc adm release extract --help
+
+dir=$(mktemp -d)
+pushd "${dir}"
+cp ${CLUSTER_PROFILE_DIR}/pull-secret pull-secret
+oc registry login --to pull-secret
+oc adm release extract --registry-config pull-secret --credentials-requests --cloud=gcp --to="${creds_requests_dir}" ${ADDITIONAL_OC_EXTRACT_ARGS} "${TESTING_RELEASE_IMAGE}"
+rm pull-secret
+popd
+
+if [[ "${EXTRACT_MANIFEST_INCLUDED}" != "true" ]] && [[ "${FEATURE_SET}" != "TechPreviewNoUpgrade" ]] &&  [[ ! -f ${SHARED_DIR}/manifest_feature_gate.yaml ]]; then
+  # workaround
+  # Bug 2035903 - One redundant capi-operator credential requests in “oc adm extract --credentials-requests”
+  # https://bugzilla.redhat.com/show_bug.cgi?id=2035903
+  echo "$(date -u --rfc-3339=seconds) - WARN: Workaround for https://bugzilla.redhat.com/show_bug.cgi?id=2035903, removing TechPreviewNoUpgrade CredentialsRequests"
+  annotation="TechPreviewNoUpgrade"
+  remove_tp_creds_requests "${creds_requests_dir}" "${annotation}" || exit 1
+fi
+
+echo "$(date -u --rfc-3339=seconds) - Extracted GCP credentials requests to directory: ${creds_requests_dir}"
+ls "${creds_requests_dir}"
+
+echo "$(date -u --rfc-3339=seconds) - Creating GCP IAM service accounts for CCO manual mode..."
+sa_suffix=${GCP_SERVICE_ACCOUNT#*@}
+for yaml_filename in $(ls -p "${creds_requests_dir}"/*.yaml | awk -F'/' '{print $NF}'); do
+  echo "$(date -u --rfc-3339=seconds) - Processing ${yaml_filename}"
+  readarray -t roles < <(yq-go r "${creds_requests_dir}/${yaml_filename}" "spec.providerSpec.predefinedRoles" | sed 's/- //g')
+  secret_name=$(yq-go r "${creds_requests_dir}/${yaml_filename}" "spec.secretRef.name")
+  secret_namespace=$(yq-go r "${creds_requests_dir}/${yaml_filename}" "spec.secretRef.namespace")
+  metadata_name=$(yq-go r "${creds_requests_dir}/${yaml_filename}" "metadata.name")
+  if [ -z "$secret_name" ] || [ -z "$secret_namespace" ] || [ -z "$metadata_name" ]; then
+    echo "$(date -u --rfc-3339=seconds) - ERROR: spec.secretRef.name, spec.secretRef.namespace or metadata.name is empty"
+    cat "${creds_requests_dir}/${yaml_filename}"
+    exit 1
+  fi
+  if [ -z "${roles:-}" ]; then
+    if [[ "${metadata_name}" =~ cloud-credential-operator ]]; then
+      echo "$(date -u --rfc-3339=seconds) - Using pre-configured custom role for cloud-credential-operator"
+      cmd="gcloud iam roles describe $(basename ${CCO_CUSTOM_ROLE}) --project=${GOOGLE_PROJECT_ID}"
```

Comment:
Suggest to introduce `run_command` function:
```
function run_command() {
    local CMD="$1"
    echo "Running Command: ${CMD}"
    eval "${CMD}"
}
```

Reply:
Updated, thanks!